src/features/import/hooks/usePreflight.ts

```diff
@@ -0,0 +1,227 @@
+import { CountryCode } from 'libphonenumber-js';
+import { useState } from 'react';
+
+import { ALERT_STATUS } from '../components/ImportDialog/Steps/ImportAlert';
+import { importResultAdd } from '../store';
+import messageIds from '../l10n/messageIds';
+import prepareImportOperations from '../utils/prepareImportOperations';
+import useFieldTitle from '../../../utils/hooks/useFieldTitle';
+import { useMessages } from 'core/i18n';
+import useOrganization from 'features/organizations/hooks/useOrganization';
+import useOrganizations from 'features/organizations/hooks/useOrganizations';
+import useTags from 'features/tags/hooks/useTags';
+import { ZetkinTag } from 'utils/types/zetkin';
+import getAddedOrgsSummary, {
+  checkAllValuesAreZero,
+  checkEmptyObj,
+} from '../utils/getAddedOrgsSummary';
+import {
+  IMPORT_ERROR,
+  PersonImport,
+  ZetkinPersonImportPostBody,
+} from '../utils/types';
+import { useApiClient, useAppDispatch, useAppSelector } from 'core/hooks';
+
+export interface Alert {
+  msg: string;
+  status: ALERT_STATUS;
+  title: string;
+}
+
+export default function usePreflight(orgId: number) {
```

Comment:
There's so much logic in this hook. Ideally I think there should be tests for it.

Reply:
Let's write some :D I guess this means i'll break out logic into separate functions, since I don't know (yet) how to make tests for hooks?

Reply:
You can do that, and it will probably be easier to test. But it's also possible to test hook. See for example `useClusteredActivities.spec.ts`.

I guess it comes down to how much logic would be possible to break out, versus how much logic is dependent on being able to use other hooks, store etc. If you can get all the data out of the store in the first couple of lines and pass it to a separate function (like `clusterEventsForWeekCalendar` that is used inside `useClusteredActivities`) then testing just that might make sense. The extra tests for the `useClusteredActivities` hook exists to test the additional logic that surrounds the clustering itself.